### PR TITLE
Top K Aggregation

### PR DESCRIPTION
**Pull request by akshaisarma**

### CODE REVIEW

File: src/main/java/com/yahoo/bullet/operations/aggregations/TopK.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+            return DEFAULT_MAX_MAP_ENTRIES;
+        }
+        return entries;
+    }
+
+    private static Number getThreshold(Map<String, Object> attributes)  {
+        if (Utilities.isEmpty(attributes)) {
+            return null;
+        }
+        return Utilities.getCasted(attributes, THRESHOLD_FIELD, Number.class);
```
Comment (NathanSpeidel) [inline src/main/java/com/yahoo/bullet/operations/aggregations/TopK.java]: I was thinking maybe we should log a warning if there is a mal-formed threshold provided?  Like a string or whatever.  I was thinking we might event want to return an error in initialize(), but that might be a little much.  So maybe just detect it here somehow (like if getCasted() returns null but the field exists in attributes) and log a warning?

File: src/main/java/com/yahoo/bullet/operations/aggregations/sketches/Sketch.java [hunk 2]
```diff
@@ -97,14 +95,14 @@ public Clip getResult(String metaKey, Map<String, String> conceptKeys) {
...
      * @param metadata The non-null {@link Map} representing the metadata.
      * @param key The key to add if not null.
      * @param supplier A {@link Supplier} that can produce a value to add to the metadata for the key. If the supplier
      *                 produces null, it is not added.
      */
-    static void addIfKeyNonNull(Map<String, Object> metadata, String key, Supplier<Object> supplier) {
+    static void addIfNonNull(Map<String, Object> metadata, String key, Supplier<Object> supplier) {
         Object data = null;
         if (key != null) {
             data = supplier.get();
```
Comment (NathanSpeidel) [inline src/main/java/com/yahoo/bullet/operations/aggregations/sketches/Sketch.java]: A few lines below this you do metadata.put(key, supplier.get())  -  want to do metadata.put(key, data) instead to save the extra function call?

File: src/main/java/com/yahoo/bullet/operations/aggregations/sketches/FrequentItemsSketch.java [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+    @Override
+    public byte[] serialize() {
+        return sketch.toByteArray(SER_DE);
+    }
+
+    @Override
+    public Clip getResult(String metaKey, Map<String, String> conceptKeys) {
+        Clip data = super.getResult(metaKey, conceptKeys);
+        ItemsSketch.Row<String>[] items = sketch.getFrequentItems(threshold, type);
```
Comment (akshaisarma) [inline src/main/java/com/yahoo/bullet/operations/aggregations/sketches/FrequentItemsSketch.java]: Yes

File: src/main/java/com/yahoo/bullet/operations/aggregations/sketches/Sketch.java (hunk 2 above)
Comment (akshaisarma) [inline src/main/java/com/yahoo/bullet/operations/aggregations/sketches/Sketch.java]: Sure

File: src/main/java/com/yahoo/bullet/operations/aggregations/TopK.java (hunk 1 above)
Comment (akshaisarma) [inline src/main/java/com/yahoo/bullet/operations/aggregations/TopK.java]: We don't do that for most things that have defaults so I chose not to. For example, aggregation size defaults to the default if it's malformed, bad, not in range etc.

Comment (NathanSpeidel) [inline src/main/java/com/yahoo/bullet/operations/aggregations/TopK.java]: Cool, maybe we can add a "warning"s that we can return in the metaData in the future or something...

**Pull request merged**